### PR TITLE
Join table

### PR DESCRIPTION
**Pull request by vic778**
in this branch I have :
- Created the owner's table
- Created the species table
- Inserted the values into my new tables as well
- Updated the column 
- Selected all animals belonging to Melody
- Joined all animals that are pokemon 
- Left Joined who owns the most animals

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -1,13 +1,47 @@
...
+-- add column species_id which is a foreign key referencing species table
+ALTER TABLE animals
+  ADD species_id INT, 
+  ADD CONSTRAINT species_id
+  FOREIGN KEY(species_id)
+  REFERENCES species(id);
+
+-- Add column owner_id which is a foreign key referencing the owners table
 
+  LTER TABLE animals
```
Comment (Meltrust) [inline schema.sql]: - [ ] You have a typo here, please kindly correct it so it reads 'ALTER TABLE'.

File: queries.sql [hunk 2]
```diff
@@ -11,3 +8,79 @@ SELECT * FROM animals WHERE neutered='true';
...
 SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
 
+BEGIN;
+UPDATE animals SET species = 'unspecified';
+ROLLBACK;
+COMMIT;
+
+BEGIN;
+UPDATE animals SET species = 'Digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'Pokemon' WHERE species IS NULL;
```
Comment (Meltrust) [inline queries.sql]: - [ ] In these 2 data updates, you should SET the `species_id`, not the `species`.  In line 18, it should read `...WHERE species_id...`.



This is an example:

```SQL

UPDATE animals
   SET species_id = 'Digimon' WHERE name like '%mon';

UPDATE animals
   SET species_id = 'Pokemon' WHERE species_id IS NULL;

```

- [ ] These SETs should go into data.sql, not queries.sql.  Kindly move them.

File: data.sql [hunk 3]
```diff
@@ -1,21 +1,161 @@
...
+SET owner_id = (SELECT id FROM owners WHERE full_name = 'Sam Smith') 
+WHERE name = 'Agumon';
+
+UPDATE animals 
+SET owner_id = (SELECT id FROM owners WHERE full_name = 'Jennifer Orwell') 
+WHERE name = 'Pikachu';
+
+UPDATE animals 
+SET owner_id = (SELECT id FROM owners WHERE full_name = 'Jennifer Orwell') 
+WHERE name = 'Gabumon';
```
Comment (Meltrust) [inline data.sql:166]: - [ ] You are missing here:

1. Melody Pond's animals (Charmander, Squirtle, and Blossom).
2. Dean Winchester's animals (Angemomn and Boardmon).

Kindly implement them, as per project requirements. 💪

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi @vic778.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



### To highlight:

- Nice code indentation✔️ 

You can do it

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: data.sql [hunk 4]
```diff
@@ -1,21 +1,190 @@
...
+)
+VALUE ('Pokemon')
+
+INSERT INTO species (
+  name
+)
+VALUE ('Digimon')
+
+BEGIN;
+UPDATE animals SET species_id = 'Digimon' WHERE name LIKE '%mon';
```
Comment (SajjadAhmad14) [inline data.sql:140]: Please select the `id` from `species` where the name is `Digimon` and assign it to `species_id` whereas currently, you are assigning `Digimon` as `species_id`.

File: data.sql [hunk 5]
```diff
@@ -1,21 +1,190 @@
...
+UPDATE animals SET species_id = 'Pokemon' WHERE species_id IS NULL;
```
Comment (SajjadAhmad14) [inline data.sql:141]: Please select the `id` from `species` where the name is `Pokemon` and assign it to `species_id` whereas currently, you are assigning `Pokemon` as `species_id`.

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Victoir,
### Great work on making the changes requested by a previous reviewer 👏 
### You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.

## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
**_Note: Please do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: data.sql (hunk 5 above)
Comment (vic778) [reply data.sql:141]: hi @SajjadAhmad14 
thank you for leaving me this comment. I'm confused. I had previously done this the recent review forced me to change how I did it. now I don't know what to do.
please read the description of the project [here](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_multiple_tables.md)

**Review by SajjadAhmad14 — APPROVED** [review]
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:

### Hi Victoir, 
## Congratulations 🎉 Thank you for implementing all the requirements 🙏
### Please go ahead and merge the branch!

## Happy Coding 👏👏👏
#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**